### PR TITLE
Adapt to new GHA Runners

### PR DESCRIPTION
**Pull request by angerman**
- Explicitly install openssl on windows?

### CODE REVIEW

**Review by carbolymer — APPROVED** [review]
LGTM, thanks!

**Comment by angerman** [comment]
> LGTM, thanks! But it seems to still be failing.

Yes, this was mostly for debugging purposes. @jasagredo and @newhoggy have been figuring out the underlying openssl issue.

**Comment by angerman** [comment]
Oh!

```
ld: warning: ignoring file '/usr/local/opt/cardano/lib/libsodium.23.dylib': found architecture 'x86_64', required architecture 'arm64'
ld: warning: ignoring file '/usr/local/opt/cardano/lib/libblst.dylib': found architecture 'x86_64', required architecture 'arm64'
ld: warning: ignoring file '/usr/local/opt/cardano/lib/libsecp256k1.2.dylib': found architecture 'x86_64', required architecture 'arm64'
```

**Comment by angerman** [comment]
Sigh...

```
GHC runtime linker: fatal error: I found a duplicate definition for symbol
   __blst_platform_cap
whilst processing object file
   C:\msys64\mingw64\opt\cardano\lib\libblst.a
The symbol was previously defined in
   C:\msys64\mingw64\opt\cardano\lib\libblst.a(#2:server.o)
This could be caused by:
   * Loading two different object files which export the same symbol
   * Specifying the same object file twice on the GHCi command line
   * An incorrect `package.conf' entry, causing some object to be
     loaded twice.
ghc-9.8.1.exe: Could not load Object Code C:\msys64\mingw64\opt\cardano\lib\libblst.a(#1:assembly.o).
```
that's a GHC bug, indirectly introduced by using a more recent `libblst`. Our cross compilers were fixed for this. (Hence it doesn't happen in nix builds).

**Comment by angerman** [comment]
This is the threepenny ui failure on darwin:
```
Configuring library for threepenny-gui-0.9.4.1..
Preprocessing library for threepenny-gui-0.9.4.1..
Building library for threepenny-gui-0.9.4.1..
[ 1 of 29] Compiling Foreign.JavaScript.Include ( src/Foreign/JavaScript/Include.hs, dist/build/Foreign/JavaScript/Include.o, dist/build/Foreign/JavaScript/Include.dyn_o )
[ 2 of 29] Compiling Foreign.JavaScript.Resources ( src/Foreign/JavaScript/Resources.hs, dist/build/Foreign/JavaScript/Resources.o, dist/build/Foreign/JavaScript/Resources.dyn_o )
WARNING: /Users/runner/.ghcup/ghc/9.6.4/lib/ghc-9.6.4/bin/./ghc-9.6.4 is loading libcrypto in an unsafe way
```

File: .github/workflows/haskell.yml [hunk 1]
```diff
@@ -32,7 +32,8 @@ jobs:
         include:
           # Using include, to make sure there will only be one macOS job, even if the matrix gets expanded later on.
           # We want a single job, because macOS runners are scarce.
-          - os: macos-latest
+          # It appears that macOS latest is aarch64 M1s now.
```
Comment (smelc) [inline .github/workflows/haskell.yml:35]:
```suggestion
          # It appears that macOS latest is aarch64 M1s now, so using a fixed version, not "macos-latest".
```

Comment (angerman) [reply .github/workflows/haskell.yml]: Thank you!

**Comment by angerman** [comment]
Ok, Window 9.8 is fixed now. Windows 9.6, we need to see. That seems to randomly die :-/
I'll try to also fix macOS.

**Comment by angerman** [comment]
Alright let's try to fix this for macos-lastest as well.

**Comment by angerman** [comment]
All GHA are back to green.

**Pull request closed without merging**